### PR TITLE
Always require account activation for email domain restricted coupons

### PR DESCRIPTION
**Pull request by douglashall**
We must always require account activation before a learner can redeem
an email domain restricted coupon, otherwise anyone could create an
account using an email address with a privileged domain and redeem
the coupon.

ENT-418

### CODE REVIEW

File: ecommerce/coupons/tests/test_views.py [hunk 1]
```diff
@@ -594,6 +594,22 @@ def test_active_user_requirement_disabled(self):
         expected_url = self.get_full_url(path=reverse('basket:summary'))
         self.assert_redemption_page_redirects(expected_url)
 
+    @httpretty.activate
+    def test_active_user_email_domain_restricted_coupon_redemption(self):
+        """
+        Verify that a user must activate their account before being allowed
+        to redeem an email domain restricted coupon.
+        """
+        self.user.email = 'foo@example.com'
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: The user already has an email address. Why change it? If you need the domain...
```python
email_domain = self.user.email.split('@')[1]
```

File: ecommerce/coupons/tests/test_views.py [hunk 2]
```diff
@@ -594,6 +594,22 @@ def test_active_user_requirement_disabled(self):
...
+        self.user.save()
+        self.mock_account_api(self.request, self.user.username, data={'is_active': False})
+        self.mock_access_token_response()
+        self.create_coupon(catalog=self.catalog, code=COUPON_CODE, benefit_value=5, email_domains='example.com')
+
+        response = self.client.get(self.redeem_url_with_params())
+        self.assertEqual(response.context['course_name'], self.course.name)
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: You should ensure the correct template is rendered as well.

File: ecommerce/coupons/tests/test_views.py [hunk 3]
```diff
@@ -594,6 +594,22 @@ def test_active_user_requirement_disabled(self):
         expected_url = self.get_full_url(path=reverse('basket:summary'))
         self.assert_redemption_page_redirects(expected_url)
 
+    @httpretty.activate
+    def test_active_user_email_domain_restricted_coupon_redemption(self):
+        """
+        Verify that a user must activate their account before being allowed
+        to redeem an email domain restricted coupon.
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: domain-restricted

**Comment by douglashall** [comment]
@clintonb Anything else you think we should clean up here?

File: ecommerce/coupons/views.py [hunk 4]
```diff
@@ -169,10 +170,16 @@ def get(self, request):
         if not valid_voucher:
             return render(request, template_name, {'error': msg})
 
-        if not voucher.offers.first().is_email_valid(request.user.email):
```
Comment (clintonb) [inline ecommerce/coupons/views.py:172]: Why was this changed? The new variable isn't used anywhere else.

**Review by clintonb — APPROVED** [review]

File: ecommerce/coupons/views.py (hunk 4 above)
Comment (douglashall) [reply ecommerce/coupons/views.py:172]: I'm using `offer` on line 174 and line 182.
